tests/Serializer/ItemNormalizerTest.php

```diff
@@ -188,4 +189,34 @@ public function testDenormalizeWithIdAndUpdateNotAllowed()
         $normalizer->setSerializer($serializerProphecy->reveal());
         $normalizer->denormalize(['id' => '12', 'name' => 'hello'], Dummy::class, null, $context);
     }
+
+    public function testItDenormalizeValueObjects()
+    {
+        $context = ['resource_class' => DummyValueObject::class, 'api_allow_update' => true];
+
+        $propertyNameCollection = new PropertyNameCollection(['name']);
```

Comment:
I'm trying to understand this, but this exact same test would pass just by replacing by the following 
```php
$propertyNameCollection = new PropertyNameCollection(['foo', 'bar']);
```
what it should be if I understand this correctly, and
```php
$propertyMetadataFactoryProphecy->create(DummyValueObject::class, 'foo', [])->willReturn($propertyMetadataFactory)->shouldBeCalled();
$propertyMetadataFactoryProphecy->create(DummyValueObject::class, 'bar', [])->willReturn($propertyMetadataFactory)->shouldBeCalled();
``` 
line 203

Apologizes if I'm missing the point, sorry

Reply:
The same result is to expect. But i'll fix this little mistake :) .